main/src/mill/modules/CoursierSupport.scala

```diff
@@ -306,4 +307,32 @@ object CoursierSupport {
     }
   }
 
+  def repoFromString(str: String, origin: String): Option[Seq[Repository]] = {
+    val spaceSep = "\\s+".r
+
+    val l =
+      if (spaceSep.findFirstIn(str).isEmpty)
+        str
+          .split('|')
+          .toSeq
+          .filter(_.nonEmpty)
+      else
+        spaceSep
+          .split(str)
+          .toSeq
+          .filter(_.nonEmpty)
+
+    RepositoryParser.repositories(l).either match {
+      case Left(errs) =>
+        System.err.println(
+          s"Ignoring $origin, error parsing repositories from it:" + System.lineSeparator() +
+            errs.map("  " + _ + System.lineSeparator()).mkString
+        )
```

Comment:
Can we fail loudly if repository parsing fails, rather than logging and continuing? I think we can make `parseBuildFiles` return a `mill.eval.Result`, and then it'll be nicely shown to the user when the `parseBuildFiles` target fails without an ugly stack trace

Reply:
Good idea. Feel free to change it, I will be some hours away from my laptop.

Reply:
I changed it to return a Result to propagate the parse error.